File: python/0001.two-sum.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hash = {}
        for index, num in enumerate(nums):
            minus = target - num
            try:
                hash[minus]
                return [hash[minus], index]
            except KeyError:
                hash[num] = index

File: go/0001.two-sum.go
func twoSum(nums []int, target int) []int {
	n := len(nums)
	hash := make(map[int]int, n)
	for i := 0; i < n; i++ {
		minus := target - nums[i]
		if index, found := hash[minus]; found {
			return []int{index, i}
		}
		hash[nums[i]] = i
	}
	return []int{}
}
